dataRetriever/txpool/interface.go

```diff
@@ -10,7 +10,7 @@ type txCache interface {
 
 	AddTx(tx *txcache.WrappedTransaction) (ok bool, added bool)
 	GetByTxHash(txHash []byte) (*txcache.WrappedTransaction, bool)
-	RemoveTxByHash(txHash []byte) error
-	CountTx() uint64
+	RemoveTxByHash(txHash []byte) bool
```

Comment:
Good point. This was before this PR. We have to add `Remove() error` in the `Cacher` interface actually, but I didn't want to make this change in the PR, since @iulianpascalau is currently modifying the interface. Can I leave as it is for now, and then we'll come back?

I have the following related TODO: `In the future, add this method to the "storage.Cacher" interface`.